src/Workspaces/VisualBasic/Portable/Editing/VisualBasicImportAdder.vb

```diff
@@ -0,0 +1,90 @@
+                    If symbol IsNot Nothing Then
+                        namespaces.Add(symbol)
+                    End If
+                Next
+            Next
+        End Sub
+
+        Protected Overrides Function GetImportedNamespaceSymbol(namespaceImport As SyntaxNode, model As SemanticModel) As INamespaceSymbol
+            Select Case namespaceImport.Kind
+                Case SyntaxKind.ImportsStatement
+                    Dim import = DirectCast(namespaceImport, ImportsStatementSyntax)
+                    Return GetImportedNamespaceSymbol(import.ImportsClauses(0), model)
+                Case SyntaxKind.SimpleImportsClause
+                    Dim clause = DirectCast(namespaceImport, SimpleImportsClauseSyntax)
+                    Return TryCast(model.GetSymbolInfo(clause.Name).Symbol, INamespaceSymbol)
+                Case Else
+                    Return Nothing
+            End Select
+        End Function
+
+        Protected Overrides Function GetExplicitNamespaceSymbol(node As SyntaxNode, model As SemanticModel) As INamespaceSymbol
+            Dim qname = TryCast(node, QualifiedNameSyntax)
+            If qname IsNot Nothing Then
+                Return GetExplicitNamespaceSymbol(qname, qname.Left, model)
+            End If
+
+            Dim maccess = TryCast(node, MemberAccessExpressionSyntax)
+            If maccess IsNot Nothing Then
+                Return GetExplicitNamespaceSymbol(maccess, maccess.Expression, model)
+            End If
+
+            Return Nothing
+        End Function
+
+        Private Overloads Function GetExplicitNamespaceSymbol(fullName As ExpressionSyntax, namespacePart As ExpressionSyntax, model As SemanticModel) As INamespaceSymbol
+            ' name must refer to something that Is Not a namespace, but be qualified with a namespace.
+            Dim Symbol = model.GetSymbolInfo(fullName).Symbol
+            Dim nsSymbol = TryCast(model.GetSymbolInfo(namespacePart).Symbol, INamespaceSymbol)
+
+            If Symbol IsNot Nothing AndAlso Symbol.Kind <> SymbolKind.Namespace AndAlso nsSymbol IsNot Nothing Then
+                ' use the symbols containing namespace, And Not the potentially less than fully qualified namespace in the full name expression.
+                Dim ns = Symbol.ContainingNamespace
+                If ns IsNot Nothing Then
+                    Return model.Compilation.GetCompilationNamespace(ns)
+                End If
+            End If
+
+            Return Nothing
+        End Function
+
+        Protected Overrides Function InsertNamespaceImport(root As SyntaxNode, gen As SyntaxGenerator, import As SyntaxNode, options As OptionSet) As SyntaxNode
+            Dim comparer = If(options.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, LanguageNames.VisualBasic), ImportsStatementComparer.SystemFirstInstance, ImportsStatementComparer.NormalInstance)
+
+            ' find insertion point
+            For Each existingImport In gen.GetNamespaceImports(root)
```

Comment:
Declaring a loop variable without `As` clause may result in reusing existing field or property from an outer scope (which might be added later).